lib/expand/join.go

```diff
@@ -512,45 +538,44 @@ func (p *Peer) printStep(format string, args ...interface{}) {
 	p.dispatcher.Send(event)
 }
 
-// createExpandOperation creates a new expand operation
-func (p *Peer) createExpandOperation(operator ops.Operator, cluster ops.Site) (*ops.SiteOperation, error) {
-	key, err := operator.CreateSiteExpandOperation(p.ctx, ops.CreateSiteExpandOperationRequest{
-		AccountID:   cluster.AccountID,
-		SiteDomain:  cluster.Domain,
-		Provisioner: schema.ProvisionerOnPrem,
-		Servers:     map[string]int{p.Role: 1},
-	})
+// dialWizard connects to a wizard
+func (p *Peer) dialWizard(addr string) (*operationContext, error) {
+	ctx, err := p.connectWizard(addr)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	err = operator.SetOperationState(*key, ops.SetOperationStateRequest{
-		State: ops.OperationStateReady,
-	})
+	return ctx, nil
+}
+
+func (p *Peer) dialCluster(addr, operationID string) (*operationContext, error) {
+	ctx, err := p.connectCluster(addr)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	operation, err := operator.GetSiteOperation(*key)
+	err = p.checkAndSetServerProfile(ctx.Cluster.App)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	return operation, nil
-}
-
-// getExpandOperation returns existing expand operation created via UI
-func (p *Peer) getExpandOperation(operator ops.Operator, cluster ops.Site, operationID string) (*ops.SiteOperation, error) {
-	operation, err := operator.GetSiteOperation(ops.SiteOperationKey{
-		AccountID:   cluster.AccountID,
-		SiteDomain:  cluster.Domain,
-		OperationID: operationID,
-	})
-	if err != nil {
+	var operation *ops.SiteOperation
```

Comment:
I think that all this logic of retrieving, expand operation either by ID or from the join backend would be better off in a separate method. I know there's already `getExpandOperation`, maybe call the new one `fetchExpandOperation`.